File: www/cgi-bin/count.py
#!/usr/bin/python

import cgi
import os

def incrementcount():
    count_file_path = os.path.join(os.getcwd(), 'www/count.txt')
    try:
        with open(count_file_path, 'r+') as f:
            i = int(f.read().strip())
            i += 1
            f.seek(0)
            f.write(str(i))
            return i
    except Exception as e:
        return -1

print('Status: 200')
print('Content-type: text/html')
print("\r\n\r\n")
print('<html>')
print('<body>')
print('<h1>The current count is: {}</h1>'.format(incrementcount()))
print('<p><a href="/">Click here to return to the homepage</a></p>')
print('</body>')
print('</html>')

File: www/cgi-bin/count.pl
#!/usr/bin/perl

use strict;
use warnings;

sub incrementcount {
    my $count_file_path = 'count.txt';
    open my $fh, '+<', $count_file_path or die "Couldn't open file: $!";
    flock($fh, 2);
    my $i = <$fh>;
    $i++;
    seek($fh, 0, 0);
    print $fh "$i\n";
    close $fh;
    return $i;
}

print "Status: 200\n";
print "Content-type: text/html\n\n";
print "\r\n\r\n";
print "<html>\n";
print "<body>\n";
print "<h1>The current count is: ", incrementcount(), "</h1>\n";
print '<p><a href="/">Click here to return to the homepage</a></p>';
print "</body>\n";
print "</html>\n";
